### DISCUSSION TITLE
a newbie wants to use hawtio 3.0.1 (default build) with Camel 4 and Wildfly 29 - failed deployment...

### DISCUSSION THREAD

**Discussion by zlutor** [hawtio/hawtio#3183] [Q&A] [answered]
Hi,

@tadayosi first of all, thanks for this great tool! 

I'm newbie with Hawtio and tried to deploy it into Wildfly 29 but failed with following error message:

`{\"jboss.deployment.unit.\\\"hawtio-default-3.0.1.war\\\".undertow-deployment.UndertowDeploymentInfoService\" => \"Failed to start service
    Caused by: java.lang.NoClassDefFoundError: Failed to link io/hawt/web/servlets/JolokiaConfiguredAgentServlet (Module \\\"deployment.hawtio-default-3.0.1.war\\\" from Service Module Loader): Failed to link org/jolokia/http/AgentServlet (Module \\\"deployment.hawtio-default-3.0.1.war\\\" from Service Module Loader): javax/servlet/http/HttpServlet\"}}"`
    
I see some other error logs, too:

`Failed to define class io.hawt.web.filters.HttpHeaderFilter in Module "deployment.hawtio-default-3.0.1.war" from Service Module Loaderjava.lang.NoClassDefFoundError: Failed to link io/hawt/web/filters/HttpHeaderFilter (Module "deployment.hawtio-default-3.0.1.war" from Service Module Loader): javax/servlet/Filter
  java.base/java.lang.ClassLoader.defineClass1(Native Method)
  java.base/java.lang.ClassLoader.defineClass(Unknown Source)
  java.base/java.lang.ClassLoader.defineClass(Unknown Source)
  org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:351)
  org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:482)
  org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:276)
  org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:79)
  org.jboss.modules.Module.loadModuleClass(Module.java:765)
  org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
  org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
   ... 29 more`
   
   Would you, please, give some hints how to fix this deployment issue?
   
   Thx in advance,
   
   Zoltan

**Comment by tadayosi** [comment, marked as answer]
Hi @zlutor,

WildFly 29 is compatible with Jakarta EE 10. For containers with Jakarta EE 9+, please try Hawtio 4.0 Beta1:
https://github.com/hawtio/hawtio/releases/tag/hawtio-4.0-beta-1

While we haven't tested Hawtio with WildFly specifically so deployment may not succeed, but at least the no class def found error should be fixed with it.

**Reply by zlutor** [reply]
@tadayosi  thanks for your prompt reply, I can confirm this version can be deployed in Wildfly 29.
now I can start testing, fingers crossed everything will work, regardless the beta state!

**Reply by zlutor** [reply]
@tadayosi Hawtio works fine but reports ExchangeFailed values in volumes during traffic (Camel -> routes -> Attributes).
Could you, please, give any hints how I can figure out when and why this counter is pegged?

**Reply by tadayosi** [reply]
@zlutor That sounds like a different issue. It seems like an issue in the Camel route. But if you can file it as an issue to https://github.com/hawtio/hawtio/issues with more details on what the error is and how you reproduce it, then we can look into it further.

**Reply by zlutor** [reply]
@tadayosi  thx for your reply, meanwhile I figured out this field in hawtio report indicates some error happened with Camel exchange delivery (timeouts, exceptions, etc.). I will continue investigation in that direction...

thx again for your prompt responses!
